core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -5132,6 +5133,12 @@ private static void computeVersion(ServletContext context) {
         }
     }
 
+    @Restricted(NoExternalUse.class)
+    public boolean shouldShowStackTrace() {
+        // Used by oops.jelly
+        return Boolean.getBoolean(Jenkins.class.getName() + ".SHOW_STACK_TRACE");
+    }
```

Comment:
Given https://github.com/jenkinsci/jenkins-test-harness/pull/201/files#diff-1d6816bd6f9430280c5d8b7853286b1dR2688, I wonder whether a more classic approach (non-final field that is initialized from the system property) isn't a better approach to this.

`Jenkins.SHOW_ERROR_STACK_TRACES = true;` would be nicer over there.

Reply:
Also https://github.com/jenkinsci/jenkins/pull/4389#pullrequestreview-359391807

Reply:
See this comment: https://github.com/jenkinsci/jenkins/pull/4389#discussion_r369716605 . One approach would be to make it private, but that makes testing more difficult. On a different discussion Stephen Connolly presented an elaborate way to deal with that problem.

It's going to be difficult to have jenkins-test-harness do `Jenkins.SHOW_ERROR_STACK_TRACES = true;` because then JTH will depend upon a new Jenkins version and the new Jenkins needs the new JTH released for the tests to pass.

Reply:
Checking existence of the field in JTH using reflection will cover both the previous Jenkins version without that field and the new one with it, being private.
(IIUC Java 11 supports this as long as it's not a final keyword removal :trollface:)

> On a different discussion Stephen Connolly presented an elaborate way to deal with that problem.

Could you provide a link or at least a summary?